src/sync/utils.rs

```diff
@@ -29,12 +31,30 @@ pub(super) fn get_size_and_rows(batches: &[RecordBatch]) -> (usize, usize) {
 // output batch (meaning the last NewPk and Value role columns and the last Value column where the
 // accompanying Changed field was `true`).
 pub(super) fn squash_batches(
-    sync_schema: &SyncSchema,
+    sync_schema: &mut SyncSchema,
     data: &[RecordBatch],
-) -> Result<RecordBatch> {
-    // Concatenate all the record batches into a single one
-    let schema = data.first().unwrap().schema();
+) -> SyncResult<RecordBatch> {
     debug!("Concatenating {} batch(es)", data.len());
+    // Concatenate all the record batches into a single one, after making all fields nullable first.
```

Comment:
There are several alternatives to this, since data types have already been validated, and we really only want to align on nullability:

1. Add a validation step in `SyncSchema::try_new` that rejects messages where a field is not nullable
2. Align the schemas with `let schema = Arc::new(Schema::try_merge(data.iter().map(|batch| batch.schema().as_ref().clone()))?)`

The later is a more costly way to basically achieve the same thing as here (as it needs to iterate/clone all the schemas and iterate through all fields checking each one individually).

The former is a much simpler solution/implementation, though slightly reducing flexibility.